src/language-js/utils.js

```diff
@@ -26,6 +45,10 @@ const FLOW_SHORTHAND_ANNOTATION = new RegExp(
 );
 const FLOW_ANNOTATION = new RegExp(`^${NON_LINE_TERMINATING_WHITE_SPACE}*::`);
 
+/**
+ * @param {Node & PrettierEsNode} node
```

Comment:
@brodybits Do you think this is the right way? I'm really new to this.

Reply:
I am having a little trouble following some of the types in estree.d.ts. I would personally favor keeping these changes as simple as possible in this PR due to its size.

At some point I would like to try the typecheck in strict mode (set `"strict": true` in compilerOptions in `tsconfig.json`), which could be another major effort across all of the code in Prettier.

I am thinking the typecheck fixes in language-js is the largest set of changes we will need to get typecheck working throughout our source code.

Reply:
I don't have experience on typing, please fix what you think is wrong, glad I learn a lot.

Reply:
This change seems to work for me, but I am not sure whether or not it is what we want:

```diff
- * @param {Node & PrettierEsNode} node
+ * @param {PrettierEsNode} node
```

I would probably need some time to look through the types in `eslint.d.ts`, look through the functions in `util.js`, and look through the code where these functions are called to better understand what is going on with these types.

My original purpose was to simply get typecheck working on src/language-js without too many `@ts-ignore` comments. It would be ideal to get typecheck working in strict mode, and declare specific types on all functions, but I think this should be a different effort. I hope we don't have to add anything else to this PR.